framework/src/vectorpostprocessors/ExtraIDIntegralVectorPostprocessor.C

```diff
@@ -152,3 +175,37 @@ ExtraIDIntegralVectorPostprocessor::threadJoin(const UserObject & s)
     for (unsigned int i = 0; i < _volumes.size(); ++i)
       _volumes[i] += sibling._volumes[i];
 }
+
+Real
+ExtraIDIntegralVectorPostprocessor::evaluate(const ElemArg & elem,
+                                             const Moose::StateArg & libmesh_dbg_var(state)) const
+{
+  mooseAssert(state.state == 0, "We do not currently support evaluating at old states");
+  return elementValue(elem.elem);
+}
+
+Real
+ExtraIDIntegralVectorPostprocessor::evaluate(const ElemQpArg & elem,
+                                             const Moose::StateArg & libmesh_dbg_var(state)) const
+{
+  mooseAssert(state.state == 0, "We do not currently support evaluating at old states");
+  return elementValue(elem.elem);
+}
+
+Real
+ExtraIDIntegralVectorPostprocessor::elementValue(const Elem * elem) const
+{
+  mooseDoOnce(if (_spatial_evaluation_index == std::numeric_limits<unsigned int>::max()) paramError(
```

Comment:
There is something strange here. In my own model, I did not hit this error on this. I need to add a run exception test to make sure this is indeed in effect.

Reply:
More tests added.